src/components/CommentList.jsx

```diff
@@ -0,0 +1,49 @@
+// import { UserContext } from '../contexts/UserContext';
+// import { useContext } from 'react';
+import { useEffect, useState } from 'react';
+import { fetchComments } from './api';
+import { useParams } from 'react-router-dom';
+
+function CommentList(id) {
+    // const {user} = useContext(UserContext);
+    const [comments, setComments] = useState([]);
+    const {articleId} = useParams();
+    const selectedId = id
+        ? id
+        : articleId;
+    console.log(selectedId, '--selected id')
+
+    useEffect(() => {
+        if (selectedId) {
+            fetchComments(selectedId)
+            .then((commentData) => {
+                setComments(commentData);
+            });
+        }
+    }, [selectedId]);
+
+    function getComment(comment) {
```

Comment:
These functions are a little confusingly named as they are named as though they fetch/get data. Also you could extract the logic to react components to do this returning of HTML ie have a CommentCard component.